Marlin/src/pins/pins_MKS_SGEN_L.h

```diff
@@ -0,0 +1,305 @@
+//
+#define TEMP_0_PIN         0   // A0 (TH1)
+#define TEMP_BED_PIN       1   // A1 (TB)
+#define TEMP_1_PIN         2   // A2 (TH2)
+
+//
+// Heaters / Fans
+//
+#define HEATER_BED_PIN     P2_05
+#define HEATER_0_PIN       P2_07
+#define HEATER_1_PIN       P2_06
+#ifndef FAN_PIN
+  #define FAN_PIN          P2_04
+#endif
+
+
+/*
+|                _____                                            _____
+| (BEEPER) 1.31 | · · | 1.30 (BTN_ENC)          (MISO)       0.8 | · · | 0.7  (SD_SCK)
+| (LCD_EN) 0.18 | · · | 0.16 (LCD_RS)           (BTN_EN1)   3.25 | · · | 0.28 (SD_CS2)
+| (LCD_D4) 0.15 | · · | 0.17 (LCD_D5)           (BTN_EN2)   3.26 | · · | 1.20 (SD_MOSI)
+| (LCD_D6)  1.0 | · · | 1.22 (LCD_D7)           (SD_DETECT) 0.27 | · · | RST
+|           GND | · · | 5V                                   GND | · · | NC
+|                ￣￣                                             ￣￣
+|                EXP1                                             EXP2
+*/
+
+#if ENABLED(ULTRA_LCD)
+  #define BEEPER_PIN       P1_31
+  #define BTN_ENC          P1_30
+
+  #if ENABLED(CR10_STOCKDISPLAY)
+    #define LCD_PINS_RS    P1_0
+
+    #define BTN_EN1        P0_18
+    #define BTN_EN2        P0_15
+
+    #define LCD_PINS_ENABLE P1_22
+    #define LCD_PINS_D4    P0_17
+
+  #else
+    #define LCD_PINS_RS    P0_16
+
+    #define BTN_EN1        P3_25
+    #define BTN_EN2        P3_26
+
+    #define LCD_PINS_ENABLE P0_18
+    #define LCD_PINS_D4    P0_15
+
+    #define LCD_SDSS       P0_28
+    #define SD_DETECT_PIN  P0_27
+
+    #if ENABLED(FYSETC_MINI_12864)
+      #define DOGLCD_CS    P0_18
+      #define DOGLCD_A0    P0_16
+      #define DOGLCD_SCK   P0_7
+      #define DOGLCD_MOSI  P1_20
+      #define FORCE_SOFT_SPI
+
+      #define LCD_BACKLIGHT_PIN -1
+
+      #define FORCE_SOFT_SPI      // Use this if default of hardware SPI causes display problems
+                                  //   results in LCD soft SPI mode 3, SD soft SPI mode 0
+
+      #define LCD_RESET_PIN P0_15   // Must be high or open for LCD to operate normally.
+
+      #if EITHER(FYSETC_MINI_12864_1_2, FYSETC_MINI_12864_2_0)
+        #ifndef RGB_LED_R_PIN
+          #define RGB_LED_R_PIN P0_17
+        #endif
+        #ifndef RGB_LED_G_PIN
+          #define RGB_LED_G_PIN P1_0
+        #endif
+        #ifndef RGB_LED_B_PIN
+          #define RGB_LED_B_PIN P1_22
+        #endif
+      #elif ENABLED(FYSETC_MINI_12864_2_1)
+        #define NEOPIXEL_PIN    P0_17
+      #endif
+
+    #else // !FYSETC_MINI_12864
+
+      #if ENABLED(MKS_MINI_12864)
+        #define DOGLCD_CS  P0_17
+        #define DOGLCD_A0  P1_0
+      #endif
+
+      #if ENABLED(ULTIPANEL)
+        #define LCD_PINS_D5 P0_17
+        #define LCD_PINS_D6 P1_0
+        #define LCD_PINS_D7 P1_22
+      #endif
+
+    #endif // !FYSETC_MINI_12864
+
+  #endif
+
+#endif // ULTRA_LCD
+
+#if !ANY(LPC_SD_LCD, LPC_SD_ONBOARD, LPC_SD_CUSTOM_CABLE)
+  #undef USB_SD_DISABLED
+  #define USB_SD_ONBOARD
+  #define LPC_SD_ONBOARD
+#endif
```

Comment:
These options are going away. See #14325 and the other LPC1768 pins files for the updated methodology.